vkcs/internal/services/networking/v2/routers/requests.go

```diff
@@ -19,8 +29,34 @@ func (opts RouterCreateOpts) ToRouterCreateMap() (map[string]interface{}, error)
 }
 
 func Create(c *gophercloud.ServiceClient, opts routers.CreateOptsBuilder) routers.CreateResult {
-	r := routers.Create(c, opts)
-	r.Err = util.ErrorWithRequestID(r.Err, r.Header.Get(util.RequestIDHeader))
+	retryableErrors := []inetworking.ExpectedNeutronError{
+		{
+			ErrCode: 404,
+		},
+		{
+			ErrCode: 409,
+			ErrType: inetworking.NeutronErrTypeDBObjectDuplicateEntry,
+		},
+	}
+
+	ctx := context.Background()
+	if c.Context != nil {
+		ctx = c.Context
+	}
+
+	var r routers.CreateResult
+	createFunc := func() error {
+		r = routers.Create(c, opts)
+		return r.Err
+	}
+
+	createErr := inetworking.CreateResourceWithRetry(ctx, createFunc, retryableErrors,
+		routerCreationRetriesNum, routerCreationRetryDelay, routerCreationTimeout)
+
+	if createErr != nil {
+		r.Err = util.ErrorWithRequestID(createErr, r.Header.Get(util.RequestIDHeader))
+	}
+
```

Comment:
Is there any reason to place this code here instead of `vkcs/networking/resource_router.go`? It's not obvious, since it's quite common for a function in `requests` modules to simply send a request to create an object, without any extra logic.

Reply:
done

Reply:
done